base/operators.jl

```diff
@@ -88,7 +88,9 @@ should therefore also implement [`hash`](@ref).
 Similar to [`==`](@ref), except for the treatment of floating point numbers
 and of missing values. `isequal` treats all floating-point `NaN` values as equal
 to each other, treats `-0.0` as unequal to `0.0`, and [`missing`](@ref) as equal
-to `missing`. Always returns a `Bool` value.
+to `missing`. Always returns a `Bool` value. It can be assumed that two values which
+are `===` are also `isequal`, and that `isequal(x, y)` returns the same result as
+`isequal(y, x)`.
```

Comment:
Might be good to spell out that `isequal` must define an equivalence relation: it must be reflexive and transitive.